ui/segmented-bar.reel/segmented-bar.js

```diff
@@ -0,0 +1,259 @@
+    prepareForActivationEvents: {
+        value: function () {
+            // more performant than this.super()
+            AbstractButton.prepareForActivationEvents.call(this);
+
+            this._defineSelectionBinding();
+        }
+    },
+
+    /**
+     * Cancel bindings for perf
+     * Ensure not to cancel bindings SegmentedBar didn't define
+     *
+     * @method
+     */
+    exitDocument: {
+        value: function () {
+            this.cancelBindings();
+
+            if (this.templateObjects.repetition.getBinding("contentController")) {
+                this.templateObjects.repetition.cancelBinding("contentController");
+            }
+            if (this.templateObjects.repetition.getBinding("content")) {
+                this.templateObjects.repetition.cancelBinding("content");
+            }
+        }
+    },
+
+    /**
+     * Hack that uses repetition selection binding to trigger dispatchActionEvent
+     * due to repetition not dispatching events on its own
+     *
+     * @private
+     * @param {Object} selectedSegmentContent
+     */
+    _selectedSegmentContent: {
+        get: Function.noop,
+        set: function (selectedSegmentContent) {
+            if (selectedSegmentContent && this.isSelectionEnabled) {
+                // Inheirted AbstractControl.detail is a Dict
+                this.detail.set("data", selectedSegmentContent);
+                this.dispatchActionEvent();
+            }
+        }
+    },
+
+    /**
+     * @private
+     * @prop {String} value - "horizontal"|"vertical"
+     * @TODO: change type after this is fixed https://github.com/jsdoc3/jsdoc/issues/629
+     */
+    _orientation: {value: null},
+
+    /**
+     * Orientation of the bar will always be horizontal by default
+     * can be overridden by setting this property
+     * isn't auto-responsive due to inability to react to parent container dimension changes
+     *
+     * @param {String} orientation
+     * @returns {String}
+     */
+    orientation: {
+        get: function () {
+            return this._orientation;
+        },
+        set: function (orientation) {
+            if (this._orientation !== orientation) {
+                this._orientation = orientation;
+                this.needsDraw = true;
+            }
+        }
+    },
+
+    /**
+     * Redraw based on content or orientation change
+     *
+     * @method
+     */
+    draw: {
+        value: function () {
+            if (this.orientation === "vertical") {
+                this.templateObjects.repetition.element.style.webkitBoxOrient = "vertical";
+                this.templateObjects.repetition.element.style.webkitBoxDirection = "normal";
+                this.templateObjects.repetition.element.style.webkitFlexDirection = "column";
+                this.templateObjects.repetition.element.style.msFlexDirection = "column";
+                this.templateObjects.repetition.element.style.flexDirection = "column";
+            }
+        }
+    }
+});
```

Comment:
> nit @ line 100: Why is this function named?

Because it's the only function I created that doesn't overriding parent's. I can't help that parents' functions aren't named, but I can ensure mine are.